src/movegen/rooks.h

```diff
@@ -6,6 +6,9 @@
 #include <array>
 #include <cstdint>
 
+#include "magics/hashing.h"
+#include "magics/pext.h"
```

Comment:
I tried conditionally including the headers, but caused issues with below `...pext` and `...hashing` functions.

Is there a nicer pattern here?

Reply:
Done. `getRookMovesHashing` would also have to be wrapped in `#ifdef` logic, so removed this redundant function naming and implemented conditional in `getRookMoves` directly.

Same for `generateRookAttackTable`.

Is this an acceptable solution? We're still avoiding nasty inlining of the `#ifdef`.